frameworks/zpm.zsh

```diff
@@ -0,0 +1,48 @@
+() {
+
+local -r home_dir=${1}
+
+git clone --recursive https://github.com/zpm-zsh/zpm $home_dir/.zpm
+
+>| ${home_dir}/.zshrc <<\END
+autoload -Uz --                 \
+  git-alias-lookup              \
+  git-branch-current            \
+  git-branch-delete-interactive \
+  git-dir                       \
+  git-ignore-add                \
+  git-root                      \
+  git-stash-clear-interactive   \
+  git-stash-recover             \
+  git-submodule-move            \
+  git-submodule-remove          \
+  mkcd                          \
+  mkpw                          \
+  duration-info-precmd          \
+  duration-info-prexec          \
+  coalesce                      \
+  git-action                    \
+  git-info
+
+source ~/.zpm/zpm.zsh
+
+zpm load zimfw/environment
+zpm load zimfw/git
+zpm load zimfw/input
+zpm load zimfw/termtitle
+zpm load zimfw/utility
+zpm load zimfw/duration-info
+zpm load zimfw/git-info
+zpm load zimfw/asciiship
+zpm load zdharma/fast-syntax-highlighting
+zpm load zsh-users/zsh-completions,fpath:src
+zpm load zsh-users/zsh-autosuggestions,source:zsh-autosuggestions.zsh
+zpm load zsh-users/zsh-history-substring-search,source:zsh-history-substring-search.zsh
+
+bindkey "^[[A" history-substring-search-up
+bindkey "^[[B" history-substring-search-down
+END
+
+HOME=$home_dir zsh -lic exit
```

Comment:
Oh, I've figured it out! Zpm keep a cache in `${TMPDIR:-/tmp}/zsh-${UID:-user}` and it started working again after I did `rm -rf ${TMPDIR:-/tmp}/zsh-${UID:-user}`. I was not aware of this directory and didn't touch it before, but it must have have got into an invalid state along my local tests with zpm.

Maybe we should clean this cache during set up to be safe:

```suggestion
# Clean cache to be safe
command rm -rf ${TMPDIR:-/tmp}/zsh-${UID:-user}
HOME=$home_dir zsh -lic exit
```

Reply:
Great! I've added `rm` as you suggested.

Reply:
Oh, and I just learned about https://github.com/zpm-zsh/zpm#troubleshooting so yeah, we definitely should include that!  :- )